src/Text/Markdown/SlamDown/Halogen/Component.purs

```diff
@@ -132,36 +144,64 @@ evalSlamDownQuery e =
       pure next
 
     SDQ.SetDocument doc next → do
-      H.modify $ SDS.replaceDocument doc
+      state ← H.get
+      let newState = SDS.replaceDocument doc state
+      H.put newState
+      let formState = newState # \(SDS.SlamDownState s) -> SDS.formStateFromDocument s.document
+      SM.foldM (updatePickers config.browserFeatures) unit formState
       pure next
 
+updatePickers :: ∀ m v a
+  . BrowserFeatures
+  → Unit
+  → String
+  → SD.FormFieldP Identity a
+  → DSL v m Unit
+updatePickers {inputTypeSupported} _ label = case _ of
+  SD.TextBox t → case t of
+    SD.DateTime _ (Compose (Just (Identity v))) →
+      unless (inputTypeSupported IT.DateTimeLocal)
+      $ (H.query' cpDateTimePicker label $ setValue (Just (Right v))) >>= mustBeMounted
+    SD.Time _ (Compose (Just (Identity v))) →
+      unless (inputTypeSupported IT.DateTimeLocal)
+      $ (H.query' cpTimePicker label $ setValue (Just (Right v))) >>= mustBeMounted
+    SD.Date (Compose (Just (Identity v))) →
+      unless (inputTypeSupported IT.DateTimeLocal)
+      $ (H.query' cpDatePicker label $ setValue (Just (Right v))) >>= mustBeMounted
+    _ → pure unit
+  _ → pure unit
+  where
+  mustBeMounted ∷ ∀ s f g p o x. Maybe x → HalogenM s f g p o m x
+  mustBeMounted (Just x) = pure x
+  mustBeMounted _ = halt $ "children at label " <> label <> " must be mounted"
```

Comment:
I think it'd be better to silently fail by absorbing the `maybe` with a `for_` rather than `halt` here - I'd only generally use the `halt` technique when a component is proxying or something like that, so it's known that the child will exist permanently within the parent.